[Merged by Bors] - ci: count only merged PRs in new contributor check

jessealama opened the pull request:
Fixes #30858

The "Label New Contributors" workflow was counting all closed PRs when determining contributor experience, but it should only count merged PRs.

This PR updates the workflow to use the GitHub search API with `is:merged` to properly filter for merged pull requests only.

Now contributors are correctly identified as "new" based on their merged contributions, not just any closed PRs.

jessealama committed:
fix(CI): count only merged PRs in new contributor check

The workflow was counting all closed PRs when determining contributor
experience, but it should only count merged PRs.

This updates the workflow to use the search API with is:merged to
properly filter for merged pull requests only.

Fixes #30858

grunweg commented:
Thanks for your PR; that is a useful bug to fix! The new logic looks good to me, but I haven't fought with all the tiny details of the github API enough to be fully confident. Let me ask @bryangingechen who's one of mathlib's CI experts:

jessealama committed:
fix(CI): search for exact bors merge prefix in PR titles

Mathlib4 uses bors for merging, which closes PRs without marking them
as merged in GitHub's standard sense. Search for the exact
"[Merged by Bors] - " title prefix instead of using the is:merged
filter.

Thanks @grunweg for the tip.

jessealama committed:
docs: add test_contributor_check.sh to scripts/README.md

jessealama committed:
chore: remove test_contributor_check.sh
